docs/en/advanced_tutorials/basedataset.md

```diff
@@ -1,3 +1,503 @@
+
+2. Implement a new dataset class that inherits from the BaseDataset and overloads the `load_data_list(self):` function of the BaseDataset to handle annotation files that don't meet the specification and guarantee a return value of `list[dict]`, where each `dict` represents a data sample.
+
+## Other features of BaseDataset
+
+The BaseDataset also contains the following features:
+
+### lazy init
+
+When the BaseDataset is instantiated, the annotation file needs to be read and parsed, therefore it will take some time. However, in some cases, such as the visualization of prediction, only the meta information of the BaseDataset is required, and reading and parsing the annotation file may not be necessary. To save time on instantiating the BaseDataset in this case, the BaseDataset supports lazy init:
+
+```python
+pipeline = [
+    LoadImage(),
+    ParseImage(),
+]
+
+toy_dataset = ToyDataset(
+    data_root='data/',
+    data_prefix=dict(img_path='train/'),
+    ann_file='annotations/train.json',
+    pipeline=pipeline,
+    # Pass the lazy_init variable in here
+    lazy_init=True)
+```
+
+When `lazy_init=True`, the initialization of ToyDataset's only performs steps 1, 2, and 3 of the BaseDataset initialization process. At this time, `toy_dataset` was not fully initialized, since `toy_dataset` will not read and parse the annotation file. The `toy_dataset` only set the meta information of the dataset (`metainfo`).
+
+Naturally, if you need to access specific data information later, you can manually call the `toy_dataset.full_init()` interface to perform the complete initialization process, during which the data annotation file will be read and parsed. Calling the `get_data_info (independence idx)`, `__len__ ()`, `__getitem__ (independence idx)`, ` get_subset_ (indices)` and `get_subset(indices)` interface will also automatically call the `full_init()` interface to perform the full initialization process (only on the first call, later calls will not call the `full_init()` interface repeatedly):
+
+```python
+# Full initialization
+toy_dataset.full_init()
+
+# After initialization, you can now get the data info
+len(toy_dataset)
+# 2
+toy_dataset[0]
+# {
+#     'img_path': "data/train/xxx/xxx_0.jpg",
+#     'img_label': 0,
+#     'img': a ndarray with shape (H, W, 3), which denotes the value the image,
+#     'img_shape': (H, W, 3) ,
+#     ...
+# }
+```
+
+**Notice:**
+
+Performing full initialization by calling the `__getitem__()` interface directly carries some risks: If a dataset class is not fully initialized by setting `lazy_init=True` firstly, then it is directly sent to the dataloader. In the subsequent data reading process, different dataloader workers will read and parse the annotation file at the same time. Although this may work normally, it consumes a lot of time and memory. **Therefore, it is recommended to manually call the `full_init()` interface to perform the full initialization process before you need to access specific data.**
```

Comment:
```suggestion
Performing full initialization by calling the `__getitem__()` interface directly carries some risks: If a dataset object is not fully initialized by setting `lazy_init=True` firstly, then it is directly sent to the dataloader. Different dataloader workers will read and parse the annotation file at the same time in the subsequent data reading process. Although this may work normally, it consumes a lot of time and memory. **Therefore, it is recommended to manually call the `full_init()` interface to perform the full initialization process before you need to access specific data.**
```